core/src/components/datetime/datetime.tsx

```diff
@@ -2226,10 +2232,34 @@ export class Datetime implements ComponentInterface {
         }}
       >
         <div class="calendar-month-grid">
-          {getDaysOfMonth(month, year, this.firstDayOfWeek % 7).map((dateObject, index) => {
-            const { day, dayOfWeek } = dateObject;
-            const { el, highlightedDates, isDateEnabled, multiple } = this;
-            const referenceParts = { month, day, year };
+          {getDaysOfMonth(month, year, this.firstDayOfWeek % 7, this.showDaysOutsideCurrentMonth).map((dateObject, index) => {
+            const { day, dayOfWeek, hiddenDay } = dateObject;
+            const { el, highlightedDates, isDateEnabled, multiple, showDaysOutsideCurrentMonth } = this;
+            let _month = month;
+            let _year = year;
+            if(showDaysOutsideCurrentMonth){
+              if(hiddenDay && day !== null && day > 20) {
```

Comment:
I think you can move the `hiddenDay && day !== null` for the if above and avoid duplication